sycl/include/sycl/detail/string.hpp

```diff
@@ -0,0 +1,45 @@
+//==----------------- string.hpp - SYCL standard header file ---------------==//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+
+#pragma once
+
+namespace sycl {
+inline namespace _V1 {
+namespace detail {
+
+class string {
+  const char *str;         // used to send existing std::string to libsycl
+  char *ret_str = nullptr; // set from libsycl
```

Comment:
This needs a comment why a single field isn't enough.
Can we initialize both with `nullptr` and use `= default;` for the default ctor?

Reply:
Added comments to explain rationale of separate data members.
I think using ctor initializer explicitly using nullptr is cleaner.